src/hyperlink/_url.py

```diff
@@ -1095,22 +1170,19 @@ def from_text(cls, text):
             look like URLs are still valid URLs. As a result, this
             method only raises :class:`URLParseError` on invalid port
             and IPv6 values in the host portion of the URL.
-
         """
         um = _URL_RE.match(_textcheck('text', text))
-        try:
-            gs = um.groupdict()
-        except AttributeError:
+        if um is None:
             raise URLParseError('could not parse url: %r' % text)
+        gs = um.groupdict()
 
         au_text = gs['authority'] or u''
         au_m = _AUTHORITY_RE.match(au_text)
-        try:
-            au_gs = au_m.groupdict()
-        except AttributeError:
+        if au_m is None:
             raise URLParseError(
                 'invalid authority %r in url: %r' % (au_text, text)
             )
+        au_gs = au_m.groupdict()
```

Comment:
If `au_m` isn't `None`, it shouldn't raise `AttributeError` here, so this now just checks if `au_m` is `None`. That avoids some casting.